fhir-term/src/main/java/com/ibm/fhir/term/util/CodeSystemSupport.java

```diff
@@ -445,6 +456,65 @@ public static Element toElement(String value, PropertyType type) {
         }
     }
 
+    /**
+     * Convert the {@link DateTime} value to a Long value that is compatible with the graph schema.
+     *
+     * @param dateTime
+     *     the dateTime value
+     * @return
+     *     the Long equivalent value (milliseconds from the epoch)
+     */
+    public static Long toLong(DateTime dateTime) {
+        TemporalAccessor value = dateTime.getValue();
+        if (value instanceof ZonedDateTime) {
+            ZonedDateTime zonedDateTime = (ZonedDateTime) value;
+            return zonedDateTime.toInstant().toEpochMilli();
+        }
+        if (value instanceof LocalDate) {
+            LocalDate localDate = (LocalDate) value;
+            return localDate.atStartOfDay().atZone(ZoneOffset.UTC).toInstant().toEpochMilli();
+        }
+        if (value instanceof YearMonth) {
+            YearMonth yearMonth = (YearMonth) value;
+            return yearMonth.atDay(1).atStartOfDay().atZone(ZoneOffset.UTC).toInstant().toEpochMilli();
+        }
+        if (value instanceof Year) {
+            Year year = (Year) value;
+            return year.atMonth(1).atDay(1).atStartOfDay().atZone(ZoneOffset.UTC).toInstant().toEpochMilli();
+        }
+        throw new IllegalArgumentException();
+    }
+
+    /**
+     * Convert the given element value to an object value that is compatible with the graph schema.
+     *
+     * @param value
+     *     the element value
+     * @return
+     *     an object value that is compatible with the graph schema
+     */
+    public static Object toObject(Element value) {
+        if (value.is(FHIR_BOOLEAN)) {
+            return value.as(FHIR_BOOLEAN).getValue();
+        }
+        if (value.is(Code.class)) {
+            return value.as(Code.class).getValue();
+        }
+        if (value.is(DateTime.class)) {
+            return DateTime.PARSER_FORMATTER.format(value.as(DateTime.class).getValue());
+        }
+        if (value.is(Decimal.class)) {
+            return value.as(Decimal.class).getValue().doubleValue();
+        }
+        if (value.is(FHIR_INTEGER)) {
+            return value.as(FHIR_INTEGER).getValue();
+        }
+        if (value.is(FHIR_STRING)) {
+            return value.as(FHIR_STRING).getValue();
+        }
+        throw new IllegalArgumentException();
+    }
```

Comment:
Is it useful outside of `fhir-term-graph`?  If so, maybe update the javadoc.  If not, then maybe it belongs back in `fhir-term-graph`.

Reply:
Updated Javadoc